### DISCUSSION TITLE
Auto-Clicker

### DISCUSSION THREAD

**Discussion by EnkerZan** [diasurgical/DevilutionX#2280] [Ideas]
I heard about this project from my buddy of mine, this is phenomenal thanks for making it. One thing I would like to see added as an option is a in-built auto-clicker. There used to be clicker tools that worked for the game, but they are incompatible for DevX. Having one implemented would be a godsend.

**Comment by qndel** [comment]
It's definitely planned, the problem is implementing it in a way that doesn't conflict with everything else :D

**Comment by AJenbo** [comment]
There are plenty of generic auto-clickers that you can use, the plan is to implement sustained attack instead.

**Reply by EnkerZan** [reply]
I've tried many and they either don't work properly or not at all. Still, something is better than nothing.

**Comment by julealgon** [comment]
I think naming it "auto-clicker" is a huge mistake. Implementing the "continuous attacking" as some sort of constant clicking simulation is a terrible way to go about it.

**Reply by EnkerZan** [reply]
As long as it makes my attacks continuous at the best possible speed while holding down the button, I'm good.

**Comment by qndel** [comment]
https://github.com/diasurgical/devilutionX/pull/2335 feel free to give this a try, it handles walking / interacting with objects / spellcasting / attacking with a single click

**Reply by EnkerZan** [reply]
Sure thing, how do you DL and install it? (I'm kinda new to this site still)

**Reply by AJenbo** [reply]
https://github.com/diasurgical/devilutionX/suites/3210954682/artifacts/74274467
(note this is the D2 style mouse hold from #2349 and not the auto clicker from #2335)

**Reply by EnkerZan** [reply]
I tried out what was in that link, works great! Only one problem so far, it broke Leoric's Tomb. As soon as I killed one of his spawned skeletons they all froze in place with a trizzed out animation. What's funny is if you kill them and walk out of the room and back in, they are alive again but frozen and you can still kill them and get EXP and drops. lol

**Reply by FluffyQuack** [reply]
You don't happen to have a save where it happens? Either way, I'm doubtful this is related to the "hold click to attack". It's very unlikely to affect enemy behaviour.

**Reply by FluffyQuack** [reply]
I did a test and I was able to reproduce it. It looks like it is caused by my PR. It has a pretty weird effect as the player can highlight multiple enemies as if they're one, and those enemies all become stuck.

Edit: Wasn't related to my PR after all, and I've made fix for the bug: https://github.com/diasurgical/devilutionX/pull/2374

**Reply by qndel** [reply]
https://github.com/diasurgical/devilutionX/suites/3207083150/artifacts/74168989 try this ;)
